### PR TITLE
Change test_bash_actions script variable to string

### PR DESCRIPTION
**Pull request by joannfeng**

### CODE REVIEW

File: test/test_dag.py [hunk 1]
```diff
@@ -299,11 +283,8 @@ def test_undetermined(clear_now_and_after):
...
 depends: 'd.txt'
 output: 'e.txt'
-
-    ''')
-    dag = get_initial_dag(graph)
-    assertDAG(
-        dag, '''
+''')
+, '''
 strict digraph "" {
```
Comment (BoPeng) [inline test/test_dag.py:288]: Can the DAGs be indented as well? You might have to add `strip` the strings in the `assertDAG` function.

Comment (joannfeng) [reply test/test_dag.py:288]: It seems `strip` is causing syntax and parsing errors. I looked it up and I think `inspect.cleandoc` or `textwrap.dedent` may work instead, so I will try using experimenting with those.

Comment (BoPeng) [reply test/test_dag.py:288]: yes, `dedent` should work. I was talking about `strip()` after splitting the strings.

**Comment by joannfeng** [comment]
I ran the file through yapf but I don't know if the indentation is standardized. Please take a look and let me know if you have any particular preferences for formatting the DAGs, and I will use `dedent` on the other DAGs as well.

**Comment by BoPeng** [comment]
Looks good, I would move `textwrap.dedent` inside `get_initial_dag` to remove redundancy though.

**Comment by BoPeng** [comment]
Unfortunately some tests fail. I think I made a mistake in converting `test_sh` before. The original tests were

```
    def test_bash(self):
        '''Test action bash'''
        script = SoS_Script(r'''
[0]
bash:
echo 'Echo'
''')
        wf = script.workflow()
        Base_Executor(wf).run()
        script = SoS_Script(r'''
[0]
bash:
echo 'Echo
''')
        wf = script.workflow()
        self.assertRaises(Exception, Base_Executor(wf).run)
```

```
    def test_sh(self):
        '''Test action run'''
        script = SoS_Script(r'''
[0]
sh:
echo 'Echo'
''')
        wf = script.workflow()
        Base_Executor(wf).run()
        script = SoS_Script(r'''
[0]
sh:
echo 'Echo
''')
        wf = script.workflow()
        self.assertRaises(Exception, Base_Executor(wf).run)
```

**Comment by joannfeng** [comment]
It seems the `execute_workflow` inside the assignment to the script variable was necessary in order to pass the tests. Also, `yapf` did not fix the lack of spaces before and after the = sign for me; I had to add those in manually. After I run `yapf`, `KeyError: 'chromium'` appears at the very bottom. Is this an issue?

**Comment by BoPeng** [comment]
No. I posted the original tests. There were two tests each and you should try to use the new style tests, even separate them. Repeated use of `execute_workflow` was wrong but that was not the one we were trying to test.

**Comment by joannfeng** [comment]
What do you mean by separate? I'm confused as to why `execute_workflow` is not to be used multiple times. Right now I have the tests like this, and they are passing: Screen Shot 2020-06-16 at 10 38 18 AM
The `execute_workflow` following `pytest.raises(Exception)` is needed to replace the `self.assertRaises` statement, but I am unsure how to replace the lines that contain `SoS_Script` and `Base_Executor` without using `execute_workflow`.

**Comment by BoPeng** [comment]
Right now for each test there are two tests, one pass and one fail. The pass one should be a single `execute_workflow`, the failed one should be `pytest.raises + execute_workflow`, and you can separate them into two tests. What you have committed has `script = execute_workflow + pytest.raises + execute_workflow(script)` with script being the return value of `execute_workflow`, which is wrong.

**Comment by joannfeng** [comment]
Okay that makes sense, thanks. Could you take a look at my newest commit and let me know if this is what you meant?

**Comment by BoPeng** [comment]
Yes, although you could have moved the `script = ` thing inside `pytest.raises`. Let us see if at least AppVeyor passes.

**Comment by BoPeng** [comment]
Thanks. I am merging since appveyor has passed. I will try to fix travis ci.

**Comment by BoPeng** [comment]
Please check https://github.com/vatlab/sos/commit/cc2a8cfaaad0cefb882ca5ebeb2d5650ea6601a1 for some further changes. The `shutil.which` part should be `pytest.mark.skipif` and you forgot the `clear_now_and_after` part.

**Comment by BoPeng** [comment]
@joannfeng travis CI is fixed. All tests should pass for future commits.

**Comment by joannfeng** [comment]
Great! Do you have any specific tests you wanted me to work on next? I was going to use `dedent` in test_execute.py, or try to pull more tests from test_dag.py out of the TestDAG class.

**Comment by BoPeng** [comment]
Does not matter. You can browse through the test and work on the easy ones or complete `test_dag`.

**Pull request merged**